packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts

```diff
@@ -207,22 +224,79 @@ class StatefulGraphChatListClient implements StatefulClient<GraphChatListClient>
   /**
    * Handle ChatListEvent event types.
    */
-  private notifyChatMessageEventChange(message: ChatListEvent) {
+  private notifyChatMessageEventChange(event: ChatListEvent) {
     this.notifyStateChange((draft: GraphChatListClient) => {
-      if (message.type === 'chatRenamed' && message.message.eventDetail) {
-        const eventDetail = message.message.eventDetail as EventMessageDetail;
-        const chatThread = draft.chatThreads.find(c => c.id === message.message.chatId);
-        if (chatThread) {
-          chatThread.topic = eventDetail.chatDisplayName;
-        }
-      }
-
-      if (message.type === 'chatMessageReceived') {
-        const chatThread = draft.chatThreads.find(c => c.id === message.message.chatId);
-        if (chatThread) {
-          const msgInfo = message.message as ChatMessageInfo;
-          chatThread.lastMessagePreview = msgInfo;
+      // find the chat thread
+      const chatThreadIndex = draft.chatThreads.findIndex(c => c.id === event.message.chatId);
+      const chatThread = draft.chatThreads[chatThreadIndex];
+
+      // func to bring the chat thread to the top of the list
+      const bringToTop = (newThread?: GraphChat) => {
+        draft.chatThreads.splice(chatThreadIndex, 1);
+        draft.chatThreads.unshift(newThread ?? chatThread);
+      };
+
+      // handle the events
+      if (event.type === 'chatRenamed' && event.message?.eventDetail && chatThread) {
+        // rename the chat thread in-place
+        chatThread.topic =
+          (event.message.eventDetail as ChatRenamedEventMessageDetail)?.chatDisplayName ?? chatThread.topic;
+      } else if (event.type === 'memberAdded' && chatThread) {
+        // inject a chat thread with only the id to force a reload; this is necessary because the
+        // notification does not include the displayName of the user who was added
+        const newThread = { id: chatThread.id } as GraphChat;
+        bringToTop(newThread);
+      } else if (event.type === 'memberRemoved' && event.message?.eventDetail && chatThread) {
+        // update the user list to remove the user; while we could add a "User removed" message
+        // the Teams client doesn't show a message and the Graph API does not show a message either
+        const membersToRemove = (event.message.eventDetail as MembersDeletedEventMessageDetail).members;
+        if (membersToRemove) {
+          const membersToRemoveSet = new Set(membersToRemove.map(member => member.id));
+          chatThread.members = chatThread.members?.filter(member => {
+            const user = member as AadUserConversationMember;
+            return !membersToRemoveSet.has(user.userId);
+          });
         }
+      } else if (event.type === 'teamsAppInstalled' && event.message?.eventDetail && chatThread) {
```

Comment:
actually, I am not familiar with this teamsAppInstalled event is about. Why we care and add a chat thread for it?

Reply:
If I add a bot to the group chat for instance, then this would trigger and it might be nice to know someone did that.